.one-pipeline.yaml

```diff
@@ -88,24 +88,57 @@ containerize:
     PIPELINE_PASSWORD=$(get_env ibmcloud-api-key-staging)  
     CONTAINER_REGISTRY=$(get_env CONTAINER_REGISTRY)
     REGISTRY_REPO=$(get_env REGISTRY_REPO)
-    IMAGE_TAG=$(get_env IMAGE_TAG)
-    IMAGE=$CONTAINER_REGISTRY/$REGISTRY_REPO:$IMAGE_TAG
+    RELEASE_TARGET=$(get_env IMAGE_TAG)
+    SAVE_IMAGE=$IMAGE
+    IMAGE=$CONTAINER_REGISTRY/$REGISTRY_REPO:$RELEASE_TARGET
+    ARCH=$(get_env architecture)
 
-    cd ../"$(load_repo app-repo path)"
-    mvn package
+    # # build P and Z
+    # if [[ "$arch" == "ZXP" ]]; then
+    #   echo "<ciorchestrator> Sending request to build P and Z"
+    #   ./scripts/pipeline/request-ciorchestrator.sh --command "make build-sample REGISTRY=${CONTAINER_REGISTRY}" --user "$W3_USERNAME" --password "$W3_PASSWORD" --branch "$RELEASE_TARGET" --repository "sample-getting-started" --org "OpenLiberty" --trigger "sample-liberty" --configFile ".ci-orchestrator/sample-liberty-build.yml" 
+    #   pipelineid=$(cat ciorchestrator-submit.id)
+    # fi
+    make build-sample
+    # preferably next 6 instructions ported to Makefile
+    # cd ../"$(load_repo app-repo path)"
+    # mvn package
     
-    docker pull icr.io/appcafe/open-liberty:kernel-slim-java11-openj9-ubi
-    docker build -t $IMAGE .
+    # docker pull icr.io/appcafe/open-liberty:kernel-slim-java11-openj9-ubi
+    # docker build -t $IMAGE .
     
-    # Push image to staging
-    echo $PIPELINE_PASSWORD | docker login stg.icr.io -u "$PIPELINE_USERNAME" --password-stdin
-    docker push $IMAGE       
+    # # Push image to staging
+    # echo $PIPELINE_PASSWORD | docker login stg.icr.io -u "$PIPELINE_USERNAME" --password-stdin
+    # docker push $IMAGE  
+
+    #wait on build for P and Z  
+    # if [[ "$arch" == "ZXP" ]]; then
+    #   # wait for build ppc64le and s390x images
+    #   echo "<ciorchestrator> waiting on request to build P and Z"
+    #   ./scripts/pipeline/await-ciorchestrator.sh --user "$W3_USERNAME" --password "$W3_PASSWORD" --pipelineId "$pipelineid"
+    # fi   
 
     # Save artifacts
-    DIGEST="$(docker inspect --format='{{index .RepoDigests 0}}' "${IMAGE}" | awk -F@ '{print $2}')"
-    ARCH="amd64"
-    echo "Saving artifact $IMAGE_TAG name=$IMAGE digest=$DIGEST arch=$ARCH"
-    save_artifact $IMAGE_TAG type=image name="$IMAGE" "digest=$DIGEST" "arch=$ARCH"   
+    echo "**** Saving Artifacts ****"
+    if [[ "$arch" == "ZXP" ]]; then
+      declare -a tags=("${RELEASE_TARGET}" "${RELEASE_TARGET}-amd64" "${RELEASE_TARGET}-ppc64le" "${RELEASE_TARGET}-s390x")
+    else
+      declare -a tags=("${RELEASE_TARGET}" "${RELEASE_TARGET}-amd64")
+    fi
+    for i in "${tags[@]}"
+      IMAGE=$SAVE_IMAGE:$i
+      DIGEST="$(skopeo inspect docker://$IMAGE | grep Digest | grep -o 'sha[^\"]*')"
```

Comment:
Does this work if you havent built the image on the same machine. Im not sure how it can find it.